### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by Priya-Patel31**
* Added Login 
* Added Signup

### CODE REVIEW

File: src/context/auth/AuthContext.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  return (
+    <AuthContext.Provider
+      value={{ isUserloggedIn, token, signup, login, logout }}
+    >
+      {children}
+    </AuthContext.Provider>
+  );
+};
+const useAuth = () => {
+  return useContext(AuthContext);
```
Comment (Ankur9669) [inline src/context/auth/AuthContext.js:68]: Suggestion: you can write this in single line

File: src/index.js [hunk 2]
```diff
@@ -4,12 +4,15 @@ import "./index.css";
...
 import { BrowserRouter as Router } from "react-router-dom";
 import { makeServer } from "./server";
+import { AuthContextProvider } from "./context/auth/AuthContext";
 
 makeServer();
 ReactDOM.render(
   <React.StrictMode>
     <Router>
+      <AuthContextProvider>
       <App />
```
Comment (Ankur9669) [inline src/index.js:14]: Suggestion: You can indent it better

File: src/pages/authentication/login/Login.js [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+import {
+  FaEnvelope,
+  BsFillEyeFill,
+  BsFillEyeSlashFill,
+  BsArrowRight,
+} from "../../../assets/icons";
+import { Signin } from "../../../assets/images";
+import "../authentication.css";
+import { useAuth } from "../../../context/auth/AuthContext";
+
```
Comment (Ankur9669) [inline src/pages/authentication/login/Login.js:13]: Suggestion: You can import all files in one single file

File: src/pages/authentication/login/Login.js [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+export const Login = () => {
+  const [showPassword, setShowPassword] = useState(false);
+  const [formData, setFormData] = useState({
+    email: "",
+    password: "",
+  });
+  let navigate = useNavigate();
+  let location = useLocation();
+  let { login } = useAuth();
+  console.log(location);
```
Comment (Ankur9669) [inline src/pages/authentication/login/Login.js:23]: Can remove this console.log statement

File: src/pages/authentication/signup/Signup.js [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+  FaEnvelope,
+  BsFillEyeFill,
+  BsFillEyeSlashFill,
+  BsArrowRight,
+} from "../../../assets/icons";
+import { Signup as SignupImage } from "../../../assets/images";
+import { Link, useNavigate } from "react-router-dom";
+import "../authentication.css";
+import {useAuth} from "../../../context/auth/AuthContext";
+
```
Comment (Ankur9669) [inline src/pages/authentication/signup/Signup.js:12]: Same here you can import all files in single file

**Review by Ankur9669 — APPROVED** [review]
The code looks fine to me, although the preview link is broken i guess you can check the link once before merging the PR

File: src/utils/fetchApi.js [hunk 6]
```diff
@@ -1,24 +1,30 @@
...
+export async function fetchApi({url, method, body, propertyName}) {
+  const user = JSON.parse(localStorage.getItem("user")) ?? {};
+  try {
+    const { data, status } = await axios({
+      method,
+      url,
+      data: body,
+      headers: { authorization: user !== null ? user?.token : "" },
+    });
+    console.log(data);
```
Comment (t007rushi) [inline src/utils/fetchApi.js:11]: console to remove

File: src/pages/authentication/signup/Signup.js (hunk 5 above)
Comment (t007rushi) [reply src/pages/authentication/signup/Signup.js:12]: rightly pointed out by @Ankur9669

**Review by t007rushi — APPROVED** [review]
Try to make small chunk of codes of lines
for easy review else look for minor suggestions and
go ahead

**Pull request merged**